Assets/Scripts/Gameplay/GameplayObjects/NetworkPickUpState.cs

```diff
@@ -0,0 +1,10 @@
+using Unity.Netcode;
```

Comment:
a single component just for this seems a bit excessive? Since pickup should only be doable by players, do we need this to be modular?

Reply:
I think here I preferred composition vs. adding this to the `ServerCharacter` and `NetworkCharacterState` god classes we have. It was trivial to just add it to the PlayerAvatar prefab. If we feel like imps/boss/vandal imp should pick up items, we'd just add it to the base Character prefab.

Reply:
God classes are anti pattern yes, but why does it apply in this case here? We need to be careful about not over engineering things. You have to ask yourself "Does setting this in a separate class make things more simple overall?" You usually want to start simple, then start adding abstractions and extracting things when you start seeing patterns. 
Wouldn't the action already have access to ServerCharacter?

Reply:
Went back and added it to `NetworkCharacterState`. Fits the same pattern as `targetId`.